controllers/kubevirtmachine_controller_test.go

```diff
@@ -268,6 +285,91 @@ var _ = Describe("reconcile a kubevirt machine", func() {
 		Expect(machineContext.KubevirtMachine.Spec.ProviderID).To(BeNil())
 	})
 
+	It("should ensure deletion of KubevirtMachine garbage collects everything successfully", func() {
+		objects := []client.Object{
+			cluster,
+			kubevirtCluster,
+			machine,
```

Comment:
Why are you using machine and not machineMock?

Reply:
Because we don't test the machine object so we use the default one which behaves as in runtime.
When we need to change the behavior of it from the regular behavior or record method calls then we should opt out into the mock.
For example your comment about checking if there was a call to machine.delete.
But it does not apply to all tests.

Reply:
I think that the best behavior is to use the mock in all tests, this will make sure we test the right things and don't add test lines because we have also the real machine object included

Reply:
@nirarg It will require changing all tests and will take a long time.
I think it's better to do it either in a separate PR or change it for tests when they require changing the default functionality of the kubevirt.machine.